lib/topological_inventory/ansible_tower/messaging_client.rb

```diff
@@ -12,7 +12,13 @@ def worker_listener
       end
 
       def targeted_refresh_listener
-        @targeted_refresh_listener ||= ManageIQ::Messaging::Client.open(targeted_refresh_listener_opts)
+        @targeted_refresh_listener ||= ManageIQ::Messaging::Client.open(targeted_refresh_listener_opts).tap do |client|
+          # persistent workers by pod hostname, this will prevent rebalances.
+          client.send(:kafka_client)[:"group.instance.id"] = ENV['HOSTNAME']
```

Comment:
So even though the hostname is different between each pod ( if we have more than one scaled up ) this stil helps us?  Or after a restart the pod has a different hostname correct? - So wouldn't a rebalance automatically occur?

Reply:
Yes, a rebalance would still occur on a deploy, there is basically no way to get around that unless we set the hostname manually somehow. 

This just takes care of the workers timing out and disconnecting which is triggering a rebalance as we're seeing right now. Basically this prevents rebalancing on pod restarts due to OOM errors or due to just errors on our side.